data-migration/main.tf

```
@@ -0,0 +1,90 @@
+resource "aws_iam_role" "s3_migration_role" {
+  name               = local.role_name
+  assume_role_policy = data.aws_iam_policy_document.allow_assume_role.json
+}
+
+data "aws_iam_policy_document" "allow_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [var.config.worker_role_arn]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+data "aws_iam_policy_document" "s3_migration_policy_document" {
+  statement {
+    sid    = "ReadOnSourceBucket"
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:GetObjectTagging",
+      "s3:GetObjectVersion",
+      "s3:GetObjectVersionTagging"
+    ]
+
+    resources = [
+      var.config.source_bucket_arn,
+    "${var.config.source_bucket_arn}/*"]
```

Just a little formatting....
```suggestion
    resources = [
      var.config.source_bucket_arn,
      "${var.config.source_bucket_arn}/*"
    ]
```